src/zenml/zen_server/jwt.py

```diff
@@ -0,0 +1,156 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Authentication module for ZenML server."""
+
+from datetime import datetime, timedelta
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    cast,
+)
+from uuid import UUID
+
+import jwt
+from pydantic import BaseModel
+
+from zenml.exceptions import AuthorizationException
+from zenml.logger import get_logger
+from zenml.zen_server.utils import server_config
+
+logger = get_logger(__name__)
+
+
+class JWTToken(BaseModel):
+    """Pydantic object representing a JWT token.
+
+    Attributes:
+        user_id: The id of the authenticated User.
+        api_key_id: The id of the API key used to authenticate the user.
+        permissions: The permissions scope of the authenticated user.
+        claims: The original token claims.
+    """
+
+    user_id: UUID
+    api_key_id: Optional[UUID] = None
+    permissions: List[str]
+    claims: Dict[str, Any] = {}
```

Comment:
Have you discussed this with @schustmi  ? will we continue using jwt permissions and scopes for rbac going forward?

Reply:
I kept these here for backwards compatibility only, but they make no sense when RBAC is introduced, because the authorization is decoupled from the JWT tokens and managed internally.